src/client/components/line-chart-for-dashboard/line-chart-for-dashboard.component.js

```diff
@@ -11,87 +12,82 @@ import {
   ReferenceArea,
 } from 'recharts';
 
-const LineChartForDashboard = ({
-  tempData,
-  strokeGrid,
-  strokeAxis,
-  minColor,
-  maxColor,
-  optimalValue,
-  strokeWidthRef,
-  strokeLine,
-  strokeWidthLine,
-  ReferanceAreaColor,
-}) => {
+export default function LineChartForDashboard({
+  data,
+  materialId,
+  boundary,
+  description,
+  unit,
+  showDetailChartFunc,
+}) {
+  const { sensorsReadings } = data;
+
+  const filterByMaterialId = sensorsReadings.filter(
+    (sensor) => sensor.fk_material_id === materialId,
+  );
+
+  const formatXAxis = () =>
+    filterByMaterialId.map((sensor) => sensor.created_at.slice(0, 13));
+
   return (
-    <LineChart
-      width={280}
-      height={180}
-      data={tempData}
-      margin={{ top: 10, right: 20, bottom: 5, left: -35 }}
-    >
-      <CartesianGrid
-        strokeDasharray="3,3"
-        stroke={strokeGrid}
-        horizontal={false}
-        vertical={false}
-        fill="#fff"
-      />
-      <XAxis
-        stroke={strokeAxis}
-        dateKey="timestamp"
-        domain={['dataMin', 'dataMax']}
-        tick={{ fontSize: 8, fill: strokeAxis }}
-      />
-      <YAxis
-        stroke={strokeAxis}
-        type="number"
-        domain={['dataMin', 'dataMax']}
-        tick={{ fontSize: 8, fill: strokeAxis }}
-        fill="none"
-        ticks={[0, 5, 10, 15, 20, 25]}
-      />
-      <Tooltip />
-      <ReferenceLine y={5} stroke={minColor} strokeWidth={strokeWidthRef} />
-      <ReferenceLine
-        y={12}
-        stroke={optimalValue}
-        strokeWidth={strokeWidthRef}
-      />
-      <ReferenceLine
-        y={22}
-        stroke={maxColor}
-        label="Max"
-        strokeWidth={strokeWidthRef}
-      />
-      <Line
-        type="monotone"
-        dataKey="temp"
-        stroke={strokeLine}
-        strokeWidth={strokeWidthLine}
-        name="Temperature"
-        unit="°C"
-      />
-      <ReferenceArea
-        y1={7}
-        y2={17}
-        stroke={ReferanceAreaColor}
-        strokeOpacity={0.3}
-      />
-    </LineChart>
+    <ResponsiveContainer width="100%" height={180}>
+      <LineChart
+        data={filterByMaterialId}
+        margin={{ top: 10, right: 20, bottom: 5, left: -28 }}
+        onClick={showDetailChartFunc}
+      >
+        <CartesianGrid
+          strokeDasharray="3,3"
+          stroke="#fff"
+          horizontal={false}
+          vertical={false}
+          fill="rgba(255,255,255, 0.5)"
+        />
+        <XAxis
+          stroke="#666"
+          dataKey={formatXAxis}
+          domain={['dataMin', 'dataMax']}
+          tick={{ fill: '#666' }}
+        />
+        <YAxis
+          stroke="#666"
+          type="number"
+          domain={[boundary.minimum - 3, boundary.maximum + 3]}
```

Comment:
for charts where sensor readings numbers are not high it looks ok but if maximum is 900, 903 as end of the chart is not visible....same issue will be in the detail chart